code/game/machinery/computer/barcode.dm

```diff
@@ -0,0 +1,70 @@
+/obj/machinery/computer/barcode
+	name = "Barcode Computer"
+	desc = "Used to print barcode stickers for the cargo routing system."
+	icon = 'goon/icons/obj/barcode.dmi'
+	icon_state = "barcode_base"
+	icon_keyboard = "barcode_keyboard"
+	icon_screen = "barcode_screen"
+	clockwork = TRUE //That'd look weird.
+	var/printing = 0
+	var/list/destinations = list("Airbridge", "Cafeteria", "EVA", "Engine", "Disposals", "QM", "Catering", "MedSci", "Security") //These have to match the ones on the cargo routers for the routers to work.
+
+/obj/machinery/computer/barcode/attack_hand(var/mob/user as mob)
+	if (..(user))
+		return
+
+	var/dat = ""
+	dat += "<b>Available Destinations:</b><BR>"
+	for(var/I in destinations)
+		dat += "<b><A href='?src=\ref[src];print=[I]'>[I]</A></b><BR>"
+	user.machine = src
+	user << browse("<TITLE>Barcode Computer</TITLE><BR>[dat]", "window=bc_computer;size=400x300")
+	onclose(user, "bc_computer")
+	return
+
+/obj/machinery/computer/barcode/Topic(href, href_list)
+	if (..(href, href_list))
+		return
+
+	if (href_list["print"] && !printing)
+		printing = 1
+		playsound(src.loc, "goon/sound/machines/printer_thermal.ogg", 50, 0)
+		sleep(28)
```

Comment:
check after sleep you still exist and are not qdeleted.

A better way to do this one is just add_timer(src, 28, "print_sticker") and implement a print sticker method